ganttproject/src/main/java/net/sourceforge/ganttproject/storage/SqlProjectDatabaseImpl.kt

```diff
@@ -127,6 +139,98 @@ class SqlProjectDatabaseImpl(private val dataSource: DataSource) : ProjectDataba
   }
 }
 
+
+class SqlTaskUpdateBuilder(private val task: Task,
+                           private val dataSource: DataSource): TaskUpdateBuilder {
+  private var lastSetStep: UpdateSetMoreStep<TaskRecord>? = null
+
+  private fun nextStep(step: (lastStep: UpdateSetStep<TaskRecord>) -> UpdateSetMoreStep<TaskRecord>) {
+    lastSetStep = step(lastSetStep ?: DSL.using(SQLDialect.H2).update(TASK))
```

Comment:
> `DSL.using(SQLDialect.H2).update(TASK)`

Creating without a connection first to make it throw only in `execute()`.

The value is not stored in `lastSetStep` initially cause jOOQ doesn't allow to execute/convert to sql queries without a single `set(...)`.